node/src/main/scala/coop/rchain/node/runtime/Setup.scala

```diff
@@ -274,7 +276,8 @@ object Setup {
           conf.devMode,
           if (conf.autopropose && conf.dev.deployerPrivateKey.isDefined) triggerProposeFOpt
           else none[ProposeFunction[F]],
-          blockReportAPI
+          blockReportAPI,
+          conf.protocolServer.networkId
```

Comment:
Shard name should be supplied from configuration also (`conf.casper.shardName`).